src/CMakeLists.txt

```diff
@@ -0,0 +1,116 @@
+if(VMA_BUILD_EXAMPLE_APP_SHADERS)
+    add_subdirectory(Shaders)
+endif()
+
+if(VMA_BUILD_REPLAY_APP)
+    add_subdirectory(VmaReplay)
+endif()
+
+set(VMA_LIBRARY_SOURCE_FILES
+    VmaUsage.cpp
+)
+
+# Use folder structure as filters in project map
+foreach(FILE ${VMA_LIBRARY_SOURCE_FILES})
+    get_filename_component(PARENT_DIR "${FILE}" PATH)
+
+    string(REPLACE "/" "\\" GROUP "${PARENT_DIR}")
+
+    if("${FILE}" MATCHES ".*\\.cpp")
+        set(GROUP "Source Files\\${GROUP}")
+    elseif("${FILE}" MATCHES ".*\\.hpp")
+        set(GROUP "Header Files\\${GROUP}")
+    endif()
+
+    source_group("${GROUP}" FILES "${FILE}")
+endforeach()
+
+add_library(VulkanMemoryAllocator ${VMA_LIBRARY_SOURCE_FILES})
+
+set_target_properties(
+    VulkanMemoryAllocator PROPERTIES
+
+    CXX_EXTENSIONS OFF
+    # Use C++14
+    CXX_STANDARD 14
+    CXX_STANDARD_REQUIRED ON
+)
+
+target_include_directories(
+    VulkanMemoryAllocator
+
+    PUBLIC
+    ${PROJECT_SOURCE_DIR}/include
+)
+
+target_link_libraries(
```

Comment:
I will continue to work on this tomorrow.

Reply:
I think the only way to fix this is to expose `VMA_DYNAMIC_VULKAN_FUNCTIONS` as CMake option. Then we can check if it's defined and then we don't need to link against `Vulkan::Vulkan`. Also I think we should only make `VMA_DYNAMIC_VULKAN_FUNCTIONS` an option and not `VMA_STATIC_VULKAN_FUNCTIONS`, because they are mutually exclusive and then we don't need to check if both are defined by accident.

Reply:
Done.